validator/remote/src/integration-test/java/tech/pegasys/teku/validator/remote/typedef/OkHttpValidatorTypeDefClientTest.java

```diff
@@ -410,6 +414,33 @@ private StateValidatorData generateStateValidatorData() {
         validator);
   }
 
+  @TestTemplate
+  public void postAttesterDuties_WhenSuccess_ReturnsResponse() throws JsonProcessingException {
+    final List<AttesterDuty> duties = List.of(randomAttesterDuty(), randomAttesterDuty());
+    final AttesterDuties response =
+        new AttesterDuties(true, dataStructureUtil.randomBytes32(), duties);
+
+    final String body = serialize(response, ATTESTER_DUTIES_RESPONSE_TYPE);
+    mockWebServer.enqueue(new MockResponse().setResponseCode(SC_OK).setBody(body));
+
+    Optional<AttesterDuties> result =
+        okHttpValidatorTypeDefClient.postAttesterDuties(UInt64.ONE, IntSet.of(1, 2));
+
+    assertThat(result).isPresent();
+    assertThat(result.get()).isEqualTo(response);
+  }
+
+  private AttesterDuty randomAttesterDuty() {
+    return new AttesterDuty(
+        dataStructureUtil.randomPublicKey(),
+        dataStructureUtil.randomValidatorIndex().intValue(),
+        dataStructureUtil.randomPositiveInt(),
+        dataStructureUtil.randomPositiveInt(),
+        dataStructureUtil.randomPositiveInt(),
+        dataStructureUtil.randomPositiveInt(),
+        dataStructureUtil.randomSlot());
+  }
```

Comment:
this should probably live in `DataStructureUtil`

Reply:
That would add a dependency, I guess that doesn't matter though?

Reply:
Oh can't do that easily, it will have a circular dependency